src/test/ReactTestUtils.js

```diff
@@ -80,14 +80,35 @@ function findAllInRenderedTreeInternal(inst, test) {
  * @lends ReactTestUtils
  */
 var ReactTestUtils = {
-  renderIntoDocument: function(instance) {
+  renderIntoDocument: function(element) {
     var div = document.createElement('div');
     // None of our tests actually require attaching the container to the
     // DOM, and doing so creates a mess that we rely on test isolation to
     // clean up, so we're going to stop honoring the name of this method
     // (and probably rename it eventually) if no problems arise.
     // document.documentElement.appendChild(div);
-    return ReactDOM.render(instance, div);
+    return ReactDOM.render(element, div);
+  },
+
+  renderIntoDocumentAsync: function(element) {
+    var promise = new Promise(function(fulfill, reject) {
+      var div = document.createElement('div');
+      // None of our tests actually require attaching the container to the
+      // DOM, and doing so creates a mess that we rely on test isolation to
+      // clean up, so we're going to stop honoring the name of this method
+      // (and probably rename it eventually) if no problems arise.
```

Comment:
@sebmarkbage @spicyj @gaearon  Do you guys want to join the bikeshedding on the function name?